server_admin/topics/admin-cli.adoc

```diff
@@ -1,1320 +1,1702 @@
+====
+
+
+=== Using the Admin CLI
+
+The Admin CLI works by making HTTP requests to Admin REST endpoints. Access to them is protected and requires authentication.
+
+//Should there be a link for the REST API documentation?
+[NOTE]
+====
+Consult the Admin REST API documentation for details about JSON attributes for specific endpoints.
+====
+
+. Start an authenticated session by providing credentials, that is, logging in. You are ready to perform create, read, update, and delete (CRUD) operations.
++
+For example, on
+
+* Linux:
++
+[options="nowrap"]
+----
+$ kcadm.sh config credentials --server http://localhost:8080/auth --realm demo --user admin --client admin
+$ kcadm.sh create realms -s realm=demorealm -s enabled=true -o
+$ CID=$(kcadm.sh create clients -r demorealm -s clientId=my_client -s 'redirectUris=["http://localhost:8980/myapp/*"]' -i)
+$ kcadm.sh get clients/$CID/installation/providers/keycloak-oidc-keycloak-json
+----
++
+* Windows:
++
+[options="nowrap"]
+----
+c:\> kcadm config credentials --server http://localhost:8080/auth --realm demo --user admin --client admin
+c:\> kcadm create realms -s realm=demorealm -s enabled=true -o
+c:\> kcadm create clients -r demorealm -s clientId=my_client -s "redirectUris=[\"http://localhost:8980/myapp/*\"]" -i > clientid.txt
+c:\> set /p CID=<clientid.txt
+c:\> kcadm get clients/%CID%/installation/providers/keycloak-oidc-keycloak-json
+----
+
+. In a production environment, you access {project_name} with `https:` to avoid exposing tokens to network sniffers. If a server's certificate is not issued by one of the trusted certificate authorities (CAs) that are included in Java's default certificate truststore, prepare a [filename]`truststore.jks` file and instruct the Admin CLI to use it.
++
+For example, on:
+
+* Linux:
++
+[options="nowrap"]
+----
+$ kcadm.sh config truststore --trustpass $PASSWORD ~/.keycloak/truststore.jks
+----
++
+* Windows:
++
+[options="nowrap"]
+----
+c:\> kcadm config truststore --trustpass %PASSWORD% %HOMEPATH%\.keycloak\truststore.jks
+----
 
 
 === Authenticating
 
-When logging in with `Admin CLI` you specify a server endpoint url, and a realm. Then you specify a username,
-or alternatively you can specify only a client id, which will result in special, so called 'service account' being used. In the first case,
-a password for the specified user has to be used at login. In the latter case there is no user password - only client secret.
-Alternatively, `Signed JWT` can be used.
+When you log in with the Admin CLI, you specify a server endpoint URL and a realm, and then you specify a username. Another option is to specify only a client ID, which results in using a special "service account". When you log in using a username, you must use a password for the specified user. When you log in using a client ID, you only need the client secret, not the user password. You could also use [command]`Signed JWT` instead of the client secret.
```

Comment:
The style guide used for our documentation requires "ID" to always be capitalized, both letters. So, if we are talking about a generic "client ID," what I have written in this sentence is what we should use.

If we are talking about a specific attribute or variable, we should spell it exactly as written/required in the command, 'clientId'. We are not doing that here.

While reading this, I think we can make the section clearer. First, are these two login methods used frequently, say each about 50% of the time? Or, is one significantly more common?

If they are used with about the same frequency, we should introduce them with a phrase and put each in a bullet, like this:

There are two ways to log in using the Admin CLI:
* With a username and password - details...
* With a client ID - details...

If one is used more frequently and the other is more of a side case, I would describe the most-used option and then put the secondary option in a NOTE box.

Reply:
@tanberry @matthewhelmke The issue here is that the text in fact is not talking about a client ID as far as I understand what ID means. Client has two distinct attributes that can both be used as an ID. One is called 'id' which is a 30 chars long random string (UUID actually) and is not very practical to use. The other is called 'clientId' which is a human readable name - but a logical name, not a caption or descriptive name for display. This name does not change so it can be used as ID. We use the second one as ID in this case. Thus talking about ID rather than clientId obfuscates things. While in this particular case we don't use 'id' attribute, it is in fact used in other cases where we also refer to it as ID. Thus when we say ID we sometimes refer to 'id', and sometimes to 'clientId' attribute. That is the point I was trying to highlight here. There would be no misunderstanding if we could just use 'clientId' when we refer to 'clientId' attribute. For readability I used 'client id' in several places instead of 'clientId'. But that was apparently misleading already as it resulted in being replaced as client ID which obfuscates it even more.

Reply:
I see. I didn't understand that until now. Perhaps we should just call this 'clientId' everywhere. What do you think?

Reply:
I will use clientId unless told otherwise.